cmd/fluxd/main.go

```diff
@@ -447,8 +450,10 @@ func main() {
 		}
 		clusterVersion = "kubernetes-" + serverVersion.GitVersion
 
-		if *k8sInCluster && !httpGitURL {
-			namespace, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+		fileinfo, err := os.Stat(k8sInClusterSecretsBaseDir)
+		isInCluster := err != nil && fileinfo.IsDir()
```

Comment:
I think you meant `isInCluster := err == nil && fileinfo.IsDir()` ... curious how a change so broken could make it to master ?

Reply:
Because you didn't take part in the review :)

Now seriously, my bad. This did t get caught by the end to end tests because we generate our own ssh key there.. we should do that.

That's for reporting it and sorry for the inconvenience.